code/_core/obj/structure/totem.dm

```diff
@@ -0,0 +1,243 @@
+/obj/structure/totem
+	name = "totem"
+	desc = "None."
+	desc_extended = "A totem that will produce a buff for the caster and their allies."
+	icon = 'icons/obj/structure/totem.dmi'
+
+	var/next_fire = 0
+	var/cooldown_fire = SECONDS_TO_DECISECONDS(1)
+
+	var/mob/living/owner //the object's owner
+	var/totem_remove_time = 0
+
+	var/leveled_effect = 0
+
+	var/affecting_faction //which faction it should affect
+
+/obj/structure/totem/Finalize()
+	. = ..()
+	start_thinking(src)
+
+/obj/structure/totem/Destroy()
+	if(owner)
+		owner.totem = null
+	return ..()
+
+/obj/structure/totem/think()
+	if(world.time >= totem_remove_time)
+		Destroy()
```

Comment:
``Destroy()`` should not directly be called. ``qdel(src)`` should be called instead.